lib/policies/proxy/proxy.js

```diff
@@ -51,15 +54,49 @@ module.exports = function (params, config) {
   if (endpoint.url) {
     return function proxyHandler (req, res) {
       logger.debug(`proxying to ${endpoint.url}, ${req.method} ${req.url}`);
-      proxy.web(req, res, { target: endpoint.url });
+      proxy.web(req, res, proxyOptions);
     };
   }
 
   // multiple urls will load balance, defaulting to round-robin
   const strategy = params.strategy || 'round-robin';
-  const balancer = createStrategy(strategy, proxy, endpoint.urls);
+  const balancer = createStrategy(strategy, proxy, proxyOptions, endpoint.urls);
 
   return function balancerProxyHandler (req, res) {
     return balancer.proxy(req, res);
   };
 };
+
+// Parse endpoint URL if single URL is provided.
+// Extend proxy options by allowing and parsing keyFile and certFile.
+function parseProxyOptions (proxyOptions = {}, endpoint = {}) {
+  let parsedProxyOptions = {};
+
+  if (Object.keys(endpoint).includes('proxyOptions')) {
+    parsedProxyOptions = Object.assign(endpoint.proxyOptions, proxyOptions);
+  }
+
+  if (!Object.keys(parsedProxyOptions).includes('target')) parsedProxyOptions.target = {};
+
+  if (parsedProxyOptions.target.keyFile) {
+    parsedProxyOptions.target.key = fs.readFileSync(parsedProxyOptions.target.keyFile);
```

Comment:
@srcnix Why the difference in option names (e.g., `keyFile` vs `key`)?

Reply:
@kevinswiber `key` is the option that the proxy module wants — `keyFile` is a custom property we use to load the certificate from file before feeding the module. Does that make sense?

Reply:
@kevinswiber, exactly what @XVincentX said. Proxy accepts cert, key and ca as a certificate string. 

In my case, and I'm certain many others, the file location is a more security means of providing the certificates. An example case would be where certificates change value, such as when a PKI with short term certificates are involved.

Reply:
:+1: